### PR TITLE
Adding FPS example and documentation for select.by_descriptor

### PR DESCRIPTION
**Pull request by bapatist**
#100  and  #101 
although @Zausinator's example on iterative fitting includes FPS, it is probably also worth having individual tasks as separate examples.

### CODE REVIEW

**Comment by bernstei** [comment]
It would be very helpful if you could create a pytest (similar to #121) that ran the script and confirmed that it ran correctly, or at least returned the correct number of configs or something basic.

File: examples/select_fps/fps.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+from wfl.configset import ConfigSet, OutputSpec
+from wfl.descriptors.quippy import calc as calc_descriptors
+from wfl.select.by_descriptor import greedy_fps_conf_global as get_fps
+import yaml
+
+# Step 1: Assign descriptors to the database
+md        = ConfigSet(input_files="md.xyz")
+md_desc   = OutputSpec(output_files="md_desc.xyz")
+
+desc_dict = yaml.safe_load(open('gap_params.yaml', 'r'))
```
Comment (bernstei) [inline examples/select_fps/fps.py:16]: considered slightly better python to do
```python
with open('gap_params.yaml') as fin:
    desc_dict = yaml.safe_load(fin)
```

File: examples/select_fps/fps.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+desc_dicts = [d for d in desc_dict if 'soap' in d.keys()] # filtering out only SOAP descriptor
+for param in desc_dicts:
+    if 'average' not in param.keys():
+        param['average']= True # to create global (per-conf) descriptors instead of local (per-atom)
+
+calc_descriptors(inputs=md, outputs=md_desc, descs=desc_dicts, key='desc')
```
Comment (bernstei) [inline examples/select_fps/fps.py:22]: `md_desc = calc_descriptors(inputs....` should work, instead of having to reopen the file in the next source line.

**Comment by bernstei** [comment]
Looks good in general, thanks.  I made a couple of tiny wording changes, and a couple of coding style suggestions. And, as I said, a pytest would be nice.

**Comment by bapatist** [comment]
Thanks for suggestions. I think I have included all of them.

**Comment by bernstei** [comment]
I'm happy to merge once the tests pass.

**Pull request merged**